src/anomalib/data/base/dataset.py

```diff
@@ -35,16 +36,38 @@
 class AnomalibDataset(Dataset, ABC):
     """Anomalib dataset.
 
+    The dataset is based on a dataframe that contains the information needed by the dataloader to load each of
+    the dataset items into memory.
+
+    The samples dataframe must be set from the subclass using the setter of the `samples` property.
+
+    The DataFrame must, at least, include the following columns:
+        - `split` (str): The subset to which the dataset item is assigned (e.g., 'train', 'test').
+        - `image_path` (str): Path to the file system location where the image is stored.
+        - `label_index` (int): Index of the anomaly label, typically 0 for 'normal' and 1 for 'anomalous'.
+        - `mask_path` (str, optional): Path to the ground truth masks (for the anomalous images only).
+        Required if task is 'segmentation'.
+
+    Example DataFrame:
+        +---+-------------------+-----------+-------------+------------------+-------+
+        |   | image_path        | label     | label_index | mask_path        | split |
+        +---+-------------------+-----------+-------------+------------------+-------+
+        | 0 | path/to/image.png | anomalous | 1           | path/to/mask.png | train |
+        +---+-------------------+-----------+-------------+------------------+-------+
+
+    Note:
+        The example above is illustrative and may need to be adjusted based on the specific dataset structure.
+
     Args:
         task (str): Task type, either 'classification' or 'segmentation'
         transform (A.Compose): Albumentations Compose object describing the transforms that are applied to the inputs.
     """
 
-    def __init__(self, task: TaskType, transform: A.Compose) -> None:
+    def __init__(self, task: TaskType, transform: Transform) -> None:
```

Comment:
Do we still use the `v2.Transform` format anywhere? I personally prefer importing the transforms directly without the `v2` prefix. It would also make it easier to switch once the prefix will be deprecated.